lib/chef/azure/chefhandlers/exception_handler.rb

```diff
@@ -30,14 +31,14 @@ def report
         load_azure_env
         message = "Check log file for details...\nBacktrace:\n"
         message << Array(backtrace).join("\n")
-        report_heart_beat_to_azure(AzureHeartBeat::READY, 0, "chef-service is running properly. Chef client run failed with error- #{message}")
+        report_heart_beat_to_azure(AzureHeartBeat::NOTREADY, 1, "chef-service is running properly. Chef client run failed with error- #{message}")
       end
     end
 
     def load_run_list

```

Comment:
Why is the exception handler the right place for this method.  Probably a historical question?  Does this mean we are dependent on the gem and configuration in the Chef config adding the handler for the initial run to work?